### PR TITLE
feat: Prover CLI Scaffoldings

### PR DESCRIPTION
**Pull request by ColoCarletti**
## What ❔

This PR implements the scaffolding for a CLI to interact with the prover. Additionally, it integrates the functionality of the `tool` crate.

## Why ❔

The purpose of this CLI is to simplify the process of interacting with the prover and enhance its usability. Users can easily interact with the prover without requiring specialized insider knowledge, ensuring that it  can managed without direct database access and complex manual queries.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

**Review by ilitteri — APPROVED** [review]
LGTM

**Comment by Artemka374** [comment]
@EmilLuta Sometimes you need just to merge main, I think it is comparing protobuf configs from there

**Review by Artemka374 — APPROVED** [review]
LGTM

File: prover/prover_cli/Cargo.toml [hunk 1]
```diff
@@ -10,11 +10,23 @@ keywords.workspace = true
 categories.workspace = true
 
 [dependencies]
+tokio = { version = "1", features = ["rt-multi-thread", "macros"] }
+env_logger = "0.10"
+log = "0.4"
```
Comment (popzxc) [inline prover/prover_cli/Cargo.toml:15]: 1. Is there a reason not to use workspace deps here?
2. Why do we need `env_logger` and `log` when we use `tracing` and `vlog` (or `tracing-subscriber`, if you have to) now?

**Pull request merged**